src/include/ipc/probe.h

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+/**
+ * \brief Definitions of shifts and masks for format encoding in probe
+ *	  extraction stream
+ *
+ * Audio format from extraction probes is encoded as 32 bit value. Following
+ * graphic explains encoding.
+ *
+ * A|BBBB|CCCC|DDDD|EEEEE|FF|GG|H|I|J|XXXXXXX
+ * A - 1 bit - Specifies Type Encoding - 1 for Standard encoding
+ * B - 4 bits - Specify Standard Type - 0 for Audio, 1 for Video
+ * C - 4 bits - Specify Audio format - 0 for PCM, 1 for MP3
+ * D - 4 bits - Specify Sample Rate - value enumerating standard sample rates:
+ *				      8000 Hz		= 0x0
+ *				      11025 Hz		= 0x1
+ *				      12000 Hz		= 0x2
+ *				      16000 Hz		= 0x3
+ *				      22050 Hz		= 0x4
+ *				      24000 Hz		= 0x5
+ *				      32000 Hz		= 0x6
+ *				      44100 Hz		= 0x7
+ *				      48000 Hz		= 0x8
+ *				      64000 Hz		= 0x9
+ *				      88200 Hz		= 0xA
+ *				      96000 Hz		= 0xB
+ *				      128000 Hz		= 0xC
+ *				      176400 Hz		= 0xD
+ *				      192000 Hz		= 0xE
+ *				      none of the above = 0xF
+ * E - 5 bits - Specify Number of Channels minus 1
+ * F - 2 bits - Specify Sample Size, number of valid sample bytes minus 1
+ * G - 2 bits - Specify Container Size, number of container bytes minus 1
+ * H - 1 bit - Specifies Sample Format - 0 for Integer, 1 for Floating point
+ * I - 1 bit - Specifies Sample Endianness - 0 for LE, 1 for BE
+ * J - 1 bit - Specifies Interleaving - 0 for Channel Interleaving,
```

Comment:
if you want all data from one channel before going to the next one, we need the block size. Since it's not specified, it looks like only interleaved samples are supported. s0 ch0, s0 ch1, s1 ch0, s1 ch1, etc...

Reply:
As I've mentioned before, interface documentation specifies broader spectrum then we are able to support at the time. I can remove obsolete definitions that are outside of our reach at the moment, and eventually add these if we ever supported these.
Also, each stream's data is always sent as separate packet, so no interleaving between streams occurs, so I think that the block size could've been inferred based on data size and number of channels, hypothetically. Because we're not doing that anyway.

Reply:
ah maybe I missed something. Is the data packet of fixed size, or is this variable size to flush whatever is in the probe extraction buffer?
In the case of fixed size, yes you could supported non-interleaved data. With variable size it becomes complicated.

Reply:
Extraction occurs on every processing cycle of the pipeline, hooked up to a buffer produce callback, and a task is scheduled to flush the data every systick, or in case of 'extraction holding buffer' getting too full, after taking up certain percentage of the buffer. Either way, data within any packet always contains a multiple of a complete sample (all channels).
Nonetheless, we're not doing that at the moment, as all our pipelines run on interleaved samples. Documentation exceeds our capabilities at the moment.

Reply:
obsolete documentation removed